### PR TITLE
Headings: accessibility, hierarchy, and cosmetics of in-line `code` therein

### PR DESCRIPTION
**Pull request by StevenBlack**
Two changes in the javascript button documentation, methods section: structural and cosmetic.

**Before** we had a `h2` - `h4` hierarchy.  The `h4` was javascript code.

Changing to a `h2` - `h3` hierarchy looks kind of wonky since the `h3` is code, and it heads a single-line explanation.

**After** — in this pull request — we have an `h2` with a definition list.

Note I added a `p` tag on the `dd` elements to get a better looking bottom margin between `dt` elements in the list.

bs-button

### CODE REVIEW

**Comment by mdo** [comment]
We'd probably want the same styles on all our code headings then, no? We have those in a lot of places.

**Comment by StevenBlack** [comment]
I count 29 instances, yes.   If you prefer `code` in definition lists rather than headings (my guess: yes) then I'm on it.

` ack "h4.*\("`

    alerts.html
    39:  <h4>$().alert()</h4>
    42:  <h4>$().alert('close')</h4>
    
    carousel.html
    198:  <h4>.carousel(options)</h4>
    206:  <h4>.carousel('cycle')</h4>
    209:  <h4>.carousel('pause')</h4>
    213:  <h4>.carousel(number)</h4>
    216:  <h4>.carousel('prev')</h4>
    219:  <h4>.carousel('next')</h4>
    
    collapse.html
    223:  <h4>.collapse(options)</h4>
    231:  <h4>.collapse('toggle')</h4>
    234:  <h4>.collapse('show')</h4>
    237:  <h4>.collapse('hide')</h4>
    
    dropdowns.html
    162:  <h4>$().dropdown('toggle')</h4>
    
    modal.html
    459:  <h4>.modal(options)</h4>
    467:  <h4>.modal('toggle')</h4>
    471:  <h4>.modal('show')</h4>
    475:  <h4>.modal('hide')</h4>
    
    popovers.html
    251:  <h4>$().popover(options)</h4>
    254:  <h4>.popover('show')</h4>
    258:  <h4>.popover('hide')</h4>
    262:  <h4>.popover('toggle')</h4>
    266:  <h4>.popover('destroy')</h4>
    
    scrollspy.html
    97:  <h4>.scrollspy('refresh')</h4>
    
    tabs.html
    95:  <h4>$().tab</h4>
    
    tooltips.html
    212:  <h4>$().tooltip(options)</h4>
    215:  <h4>.tooltip('show')</h4>
    219:  <h4>.tooltip('hide')</h4>
    223:  <h4>.tooltip('toggle')</h4>
    227:  <h4>.tooltip('destroy')</h4>

**Comment by StevenBlack** [comment]
OK I hit all 29 instances. The general case has more text than `buttons.html` so I reverted `buttons.html` back to using `h4` tags.  Starting now all `h4` tags have `code` tags where appropriate.

Here's what it looks like now, using `modal.html` here as an example.

bs-modal

**Comment by patrickhlauke** [comment]
LGTM

**Comment by hnrch02** [comment]
@mdo You down with this?

**Comment by StevenBlack** [comment]
Slightly larger for you.  I think this looks really good.

codeinh4-2

**Comment by StevenBlack** [comment]
The CSS commit above covers some instances of code-in-headings not seen in the components section of the docs.

Here's a before-and-after look at code in larger headings, code that's actually in-line with some text.

bs-code-headings

**Comment by patrickhlauke** [comment]
LGTM as well. Still waiting for @mdo's blessing :)

**Comment by XhmikosR** [comment]
:+1: from me too

**Comment by cvrebert** [comment]
I am very much in favor of using `<code>` for method headings in the API docs like this.

**Comment by XhmikosR** [comment]
@StevenBlack: can you fetch and rebase?

File: docs/assets/css/src/docs.css [hunk 1]
```diff
@@ -28,6 +28,11 @@ body {
   font-weight: normal;
 }
 
+/* In-line code within headings retain the heading background */
```
Comment (XhmikosR) [inline docs/assets/css/src/docs.css]: It's `inline` and not `in-line` as far as I can tell.

Also, break one selector per line.

**Comment by XhmikosR** [comment]
@StevenBlack: Also squash the patches and take care of my previous comment please.

File: docs/assets/css/src/docs.css (hunk 1 above)
Comment (StevenBlack) [inline docs/assets/css/src/docs.css]: @XhmikosR  Confirming: `in-line` is correct usage.

Comment (XhmikosR) [inline docs/assets/css/src/docs.css]: I'd rather have it `inline`.

**Comment by StevenBlack** [comment]
Having no success squashing.

**Comment by kkirsche** [comment]
@StevenBlack `git rebase -i HEAD~13 [your-branch-name`

You'll be presented with a list of commits that say `pick [something] "commit message"`.

Change all except the first `pick` with `squash`

**Comment by cvrebert** [comment]
Looks like the rebase had some problems. I'm happy to do the squashing when merging this. Just waiting on @mdo to approve the style changes.

**Comment by XhmikosR** [comment]
Don't specify the number of commits. Just run git rebase --i
upstream/master after you have fetched upstream.

**Comment by StevenBlack** [comment]
And we do this busy-work because....?   Just curious.

**Comment by XhmikosR** [comment]
Because it's useless to have many patches for the same thing apart from
making git history bigger. Also, it's sometimes mandatory to rebase anyway
due to conflicts.

**Comment by kkirsche** [comment]
Ah. Sorry @XhmikosR I've had bad luck doing it that way. As long as it works though in the end that's what matters.

But yeah, @StevenBlack having a single commit to point to for a patch, in my opinion, can make it easier to identify where a regression or bug was introduced and trace it back to the correct issue so you don't have 15+ commits for a single patch.

Just my opinion though :smiley:

**Pull request closed without merging**